secure_message_v2/controllers/threads.py

```diff
@@ -104,3 +106,15 @@ def marked_for_deletion_by_closed_at_date(session: Session) -> None:
     date_threshold = datetime.now(timezone.utc) - timedelta(days=app.config["THREAD_DELETION_OFFSET_IN_DAYS"])
     threads_updated = query_threads_marked_for_deletion_by_closed_at_date(date_threshold, session)
     logger.info(f"{threads_updated} Threads marked for deletion")
+
+
+@with_db_session
+def set_thread_attributes(thread_id: str, payload: dict, session: Session) -> Thread:
+    if thread := query_thread_by_id(thread_id, session):
+        for key, value in payload.items():
+            if key in UPDATABLE_ATTRIBUTES:
+                setattr(thread, key, value)
+            else:
+                logger.error(f"Thread attribute {key} can not be set")
+                raise AttributeError
```

Comment:
Debatable, this currently would update no attributes if anything is wrong in the payload. I think this is the way to go, as up wouldn't necessarily want 4 out of 5 attributes updating thinking all 5 have been done

Reply:
Makes sense.